rotkehlchen/chain/ethereum/uniswap/typing.py

```diff
@@ -65,10 +118,314 @@ def serialize(self) -> Dict[str, Any]:
+        base_asset: Union[EthereumToken, UnknownEthereumToken] = (
+            UnknownEthereumToken(
+                ethereum_address=ba_address,
+                symbol=ba_symbol,
+                name=ba_name,
+                decimals=int(ba_decimals),
+            )
+            if is_ba_unknown
+            else EthereumToken(identifier=ba_symbol)
+        )
+        quote_asset: Union[EthereumToken, UnknownEthereumToken] = (
+            UnknownEthereumToken(
+                ethereum_address=qa_address,
+                symbol=qa_symbol,
+                name=qa_name,
+                decimals=int(qa_decimals),
+            )
+            if is_qa_unknown
+            else EthereumToken(identifier=qa_symbol)
+        )
+        return cls(
+            tx_hash=trade.link,
+            address=address,
+            timestamp=trade.timestamp,
+            location=trade.location,
+            base_asset=base_asset,
+            quote_asset=quote_asset,
+            trade_type=trade.trade_type,
+            amount=trade.amount,
+            rate=trade.rate,
+            fee=trade.fee,
+            fee_currency=FakeAsset(identifier=trade.fee_currency.identifier),
+            notes=trade.notes,
+        )
+
+    def get_trade_from_ammtrade(self) -> Trade:
+        """Turns an AMMTrade to a Trade
+        """
+        # Get `fee_currency` symbol and class to instantiate
+        fc_class: Union[Type[Asset], Type[FakeAsset]]
+        if self.trade_type == TradeType.SELL:
+            fc_identifier = self.base_asset.symbol
+            fc_class = (
+                Asset
+                if isinstance(self.base_asset, EthereumToken)
+                else FakeAsset
+            )
+        else:
+            fc_identifier = self.quote_asset.symbol
+            fc_class = (
+                Asset
+                if isinstance(self.quote_asset, EthereumToken)
+                else FakeAsset
+            )
+        return Trade(
+            timestamp=self.timestamp,
+            location=self.location,
+            pair=TradePair(self.pair),
+            trade_type=self.trade_type,
+            amount=self.amount,
+            rate=self.rate,
+            fee=self.fee,
+            fee_currency=fc_class(identifier=fc_identifier),  # type: ignore
+            link=self.tx_hash,
+            notes=self.notes,
+            custom_identifier=self.identifier,
+        )
+
+    @property
+    def identifier(self) -> TradeID:
+        """Formulates a unique identifier for the trade to become the DB primary key
+
+        Current `trades` table and `Trade` class (from `data_structures.py`)
+        do not support/are not compatible with:
+          - Storing the ethereum address that did the trade.
+          - Unknown ethereum tokens, and both `pair` assets are expected to be
+          known (i.e. <Asset>).
+
+        For the reasons above, a custom trade identifier that contains the
+        the previous data.
+        """
+        partial_id = (
```

Comment:
This partial id is not needed for this object. Why do you keep it? The only reason this is the id of the normal trades is that we needed a unique way to identify a trade from its parts.

For an on-chain trade the transaction_hash + the log index should be enough.